indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -0,0 +1,107 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    'use strict';
+
+    var permissionClasses = {
+        access: 'accept',
+        timetable: 'highlight',
+        edit: 'danger',
+        material: 'warning'
+    };
+
+    $.widget('indico.permissionswidget', {
+        options: {
+            event_id: null
+        },
+
+        _update: function() {
+            this.$dataField.val(JSON.stringify(this.data));
+        },
+        _renderLabel: function(principal) {
+            var $labelBox = $('<div>', {class: 'label-box'});
+            var type = principal._type;
+            if (type === 'EventRole') {
+                var $text = $('<span>', {class: 'text-normal', text: principal.name});
+                var $code = $('<span>', {
+                    class: 'role-code',
+                    text: principal.code,
+                    css: {
+                        'border-color': '#' + principal.color,
+                        'color': '#' + principal.color
+                    }
+                });
+
+                return $labelBox.append($('<span>', {class: 'flexrow f-a-center'}).append($code).append($text));
+            } else {
+                var iconClass = type === 'Avatar' || type === 'Email' ? 'icon-user' : 'icon-users';
+                var text = type === 'Avatar' ? principal.name : principal.id;
+                return $labelBox.append($('<span>', {class: 'label-icon text-normal ' + iconClass, text: text}));
+            }
+        },
+        _renderPermissions: function(principal, permissions) {
+            var $permissions = $('<div>', {class: 'permissions-box flexrow f-a-center f-self-stretch'});
+            var $permissionsList = $('<ul>').appendTo($permissions);
+            permissions.forEach(function(item) {
+                $permissionsList.append($('<li>', {class: 'i-label bold ' + permissionClasses[item]}).append(item));
+            });
+
+            var $permissionsEditBtn = $('<button>', {
+                'type': 'button',
+                'class': 'i-button text-color borderless icon-only icon-edit',
+                'data-href': build_url(Indico.Urls.EventPermissions, {confId: this.options.event_id}),
```

Comment:
You can put all those `data-` attributes inside an object, like `'data': {'href': '...', 'title': '...}`

Reply:
Apparently this is not possible.

Reply:
That's because `data: {}` sets it as jquery data, but the declarative code actually needs the attributes, so `'data-...': ...` is correct here.